pyfar/classes/audio.py

```diff
@@ -1594,14 +1608,19 @@ def _get_arithmetic_data(data, domain, cshape, matmul, audio_type):
         Signal. `np.asarray(data)` otherwise.
     """
     if isinstance(data, (Signal, TimeData, FrequencyData)):
+        data_ = data.copy()
+        # check if complex casting of any input signal is necessarry
+        if complex:
+            if isinstance(data_, (Signal, TimeData)) and not data_.complex:
+                data_.complex = True
```

Comment:
i think in PR #396 we dont allow a setter

Reply:
I think we changed that as we noted the need for it. There are now checks that make sure `complex = False` can only be set if the spectrum is symmetric and 0 Hz and Nyquist are real valued.